web/app.js

```diff
@@ -1739,14 +1741,41 @@ function webViewerPageRendered(evt) {
     });
     // It is a good time to report stream and font types.
     PDFViewerApplication.pdfDocument.getStats().then(function (stats) {
+      // Convert the stats to telemetry friendly names.
+      let newStats = {
+        streamTypes: toTelemtryNames(swapKeyValue(StreamType),
+                                     stats.streamTypes),
+        fontTypes: toTelemtryNames(swapKeyValue(FontType), stats.fontTypes),
+      };
+
       PDFViewerApplication.externalServices.reportTelemetry({
         type: 'documentStats',
-        stats,
+        stats: newStats,
       });
     });
   }
 }
 
+function toTelemtryNames(lookup, stats) {
+  // Replace the integer types with the string name from the constant.
+  let ret = [];
+  for (let key in stats) {
+    if (!stats[key]) {
+      continue;
+    }
+    ret.push(lookup[key].padEnd(3, 'X'));
```

Comment:
Why the padding? Is there a specification for how those names should look? I'm only aware of https://github.com/mozilla/pdf.js/wiki/Enumeration-Assignments-for-the-Telemetry-Histograms, is that also still relevant?

Reply:
[Labels](https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/telemetry/collection/histograms.html#labels) have to pass the regex `^[a-z][a-z0-9_]+[a-z0-9]$` (3 char minimum).